LibreNMS/Util/Debug.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/*
+ * Debug.php
+ *
+ * -Description-
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2021 Tony Murray
+ * @author     Tony Murray <murraytony@gmail.com>
+ */
+
+namespace LibreNMS\Util;
+
+use App;
+use Illuminate\Database\Events\QueryExecuted;
+use LibreNMS\DB\Eloquent;
+use Log;
+
+class Debug
+{
+    private static $debug = false;
+    private static $verbose = false;
+
+    public static function set($debug = true, bool $silence = false): bool
```

Comment:
Why not typehint debug?

Reply:
I did at first, but we are supplying this function to an on boot startup, and the first argument sent to the function is an application.  So, the options are change that, make a specific function, or make this less strict.   I went with the later and added the cast below.